### PR TITLE
Add Trusted Apps UI to console

### PR DESCRIPTION
**Pull request by ImalshaG**
This PR changes the Advanced Configurations page of Application configurations in the console app. 

Following changes are added:
- client attestation section is taken out of app-native authentication section as a seperate section.
- a new section is added for trusted app related features
- a new section added for platform specific configurations common for both client attestation and trusted app related features

Figma design: https://www.figma.com/proto/bGsPNcdkdYrZ1sdfNFTx6p/FIDO-Trusted-Apps?node-id=3-18&t=f63TAHja5dE8mtYb-1

### CODE REVIEW

File: features/admin.applications.v1/components/forms/advanced-configurations-form.tsx [hunk 1]
```diff
@@ -27,9 +27,11 @@ import {
     TestableComponentInterface
 } from "@wso2is/core/models";
 import { Field, Form, FormPropsInterface } from "@wso2is/form";
-import { GenericIcon, Heading } from "@wso2is/react-components";
+import { ConfirmationModal, GenericIcon, Heading, URLInput  } from "@wso2is/react-components";
+
```
Comment (pavinduLakshan) [inline features/admin.applications.v1/components/forms/advanced-configurations-form.tsx:31]:
```suggestion
```

File: features/admin.applications.v1/components/forms/advanced-configurations-form.tsx [hunk 2]
```diff
@@ -111,6 +115,14 @@ export const AdvancedConfigurationsForm: FunctionComponent<AdvancedConfiguration
     );
     const [ isUserClickedClientAttestationWhenDisabled, setIsUserClickedClientAttestationWhenDisabled ] =
         useState<boolean>(false);
+    const [ isFIDOTrustedAppsEnabled, setIsFIDOTrustedAppsEnabled ] = useState<boolean>(
+        config?.trustedAppConfiguration?.isFIDOTrustedApp);
+    const [ isConsentGranted, setIsConsentGranted ] = useState<boolean>(
+        config?.trustedAppConfiguration?.isConsentGranted);
```
Comment (pavinduLakshan) [inline features/admin.applications.v1/components/forms/advanced-configurations-form.tsx:121]:
```suggestion
    const [ isFIDOTrustedAppsEnabled, setIsFIDOTrustedAppsEnabled ] = useState<boolean>(
        config?.trustedAppConfiguration?.isFIDOTrustedApp
    );
    const [ isConsentGranted, setIsConsentGranted ] = useState<boolean>(
        config?.trustedAppConfiguration?.isConsentGranted
    );
```

File: features/admin.applications.v1/components/forms/advanced-configurations-form.tsx [hunk 3]
```diff
@@ -111,6 +115,14 @@ export const AdvancedConfigurationsForm: FunctionComponent<AdvancedConfiguration
...
+    const [ showFIDOConfirmationModal, setShowFIDOConfirmationModal ] = useState<boolean>(false);
+
```
Comment (pavinduLakshan) [inline features/admin.applications.v1/components/forms/advanced-configurations-form.tsx:123]:
```suggestion
```

File: features/admin.applications.v1/components/forms/advanced-configurations-form.tsx [hunk 4]
```diff
@@ -198,125 +232,171 @@ export const AdvancedConfigurationsForm: FunctionComponent<AdvancedConfiguration
...
-                (
-                    isApplicationNativeAuthenticationEnabled &&
-                    (template?.id === ApplicationManagementConstants.CUSTOM_APPLICATION ||
-                    template?.id === ApplicationManagementConstants.CUSTOM_APPLICATION_OIDC ||
-                    template?.id === ApplicationManagementConstants.MOBILE ||
-                    template?.id === ApplicationManagementConstants.TEMPLATE_IDS.get("oidcWeb"))
-                ) && (
-                    <>
+        <>
+            <Form
```
Comment (pavinduLakshan) [inline features/admin.applications.v1/components/forms/advanced-configurations-form.tsx:301]: Is this indentation change intentional?

**Comment by pavinduLakshan** [comment]
Let's add unit tests as well.

File: features/admin.applications.v1/components/forms/advanced-configurations-form.tsx [hunk 5]
```diff
@@ -198,125 +232,171 @@ export const AdvancedConfigurationsForm: FunctionComponent<AdvancedConfiguration
...
     };
 
+    /**
+     * Handle FIDO activation value with confirmation. Deactivation is not confirmed
+     */
+    const handleFIDOActivation = (shouldActivate: boolean) => {
+        shouldActivate ?
+            setShowFIDOConfirmationModal(true)
+            : setIsFIDOTrustedAppsEnabled(false);
+    };
```
Comment (Achintha444) [inline features/admin.applications.v1/components/forms/advanced-configurations-form.tsx:297]: const handleFIDOActivation = (shouldActivate: boolean) =>  setShowFIDOConfirmationModal(shouldActivate);

File: features/admin.applications.v1/components/forms/advanced-configurations-form.tsx [hunk 6]
```diff
@@ -362,195 +431,337 @@ export const AdvancedConfigurationsForm: FunctionComponent<AdvancedConfiguration
...
+                                            isClientAttestationUIDisabled &&
+                                            isUserClickedClientAttestationWhenDisabled
+                                        ) && (
+                                            <Alert severity="info" className="client-attestation-info-alert">
+                                                { t("applications:forms.advancedConfig." +
+                                                    "sections.clientAttestation.alerts.clientAttestationAlert") }
+                                            </Alert>
+                                        )
+                                    }
+
```
Comment (Achintha444) [inline features/admin.applications.v1/components/forms/advanced-configurations-form.tsx:454]:
```suggestion
```

File: features/admin.applications.v1/components/forms/advanced-configurations-form.tsx [hunk 7]
```diff
@@ -362,195 +431,337 @@ export const AdvancedConfigurationsForm: FunctionComponent<AdvancedConfiguration
...
+                                        required={ false }
+                                        readOnly={ readOnly }
+                                        value={
+                                            config?.attestationMetaData?.enableClientAttestation
+                                                ? [ "enableClientAttestation" ]
+                                                : []
+                                        }
+                                        listen={ setIsEnableClientAttestation }
+                                        data-testid={ `${ testId }-enable-client-attestation` }
+                                        data-componentid={ `${ testId }-enable-api-based-authentication` }
```
Comment (Achintha444) [inline features/admin.applications.v1/components/forms/advanced-configurations-form.tsx:470]: use `componentid` for `data-componentid`. Check other places

File: features/admin.applications.v1/components/forms/advanced-configurations-form.tsx [hunk 8]
```diff
@@ -362,195 +431,337 @@ export const AdvancedConfigurationsForm: FunctionComponent<AdvancedConfiguration
...
+                                                    null, 4)
+                                                ] : []
+                                        }
+                                        hint={ t("applications:forms." +
+                                            "advancedConfig." +
+                                            "sections.clientAttestation.fields." +
+                                            "androidAttestationServiceCredentials.hint") }
+                                        type="text"
+                                        maxLength={ 5000 }
+                                        minLength={ 30 }
```
Comment (Achintha444) [inline features/admin.applications.v1/components/forms/advanced-configurations-form.tsx:510]: put these values into a const file. Do not hard code.

File: features/admin.applications.v1/models/application.ts [hunk 9]
```diff
@@ -265,6 +276,11 @@ export interface AdvancedConfigurationsInterface {
     enableClientAttestation?: boolean;
     androidPackageName?: string;
     androidAttestationServiceCredentials?: string;
+    trustedAppConfiguration?: TrustedAppConfigurationsInterface
+    // Check if these configs are needed here
```
Comment (Achintha444) [inline features/admin.applications.v1/models/application.ts:280]: is this comment necessary ?

File: features/admin.applications.v1/components/forms/advanced-configurations-form.tsx (hunk 4 above)
Comment (ImalshaG) [reply features/admin.applications.v1/components/forms/advanced-configurations-form.tsx:298]: This indentation was due to the newly added ConfirmationModal

File: features/admin.applications.v1/components/forms/advanced-configurations-form.tsx (hunk 5 above)
Comment (ImalshaG) [reply features/admin.applications.v1/components/forms/advanced-configurations-form.tsx:294]: This if condition uses 2 different methods, therefore cannot modify as suggested

File: features/admin.applications.v1/components/forms/advanced-configurations-form.tsx (hunk 8 above)
Comment (ImalshaG) [reply features/admin.applications.v1/components/forms/advanced-configurations-form.tsx:517]: This is an existing code block which was refactored.

**Review by pavinduLakshan — APPROVED** [review]

**Review by pavinduLakshan — APPROVED** [review]

**Pull request merged**